openlibrary/catalog/add_book/__init__.py

```diff
@@ -678,6 +685,197 @@ def load_data(rec, account_key=None):
     return reply
 
 
+def update_matched_edition(
+    e: Edition, rec: dict, need_edition_save: bool, account_key: str | None = None
+) -> tuple[Edition, bool]:
+    """
+    Update `e` fields to add values from `rec` that don't currently
+    exist on `edition`. The default is to prefer existing values rather than
+    to overwrite.
+    """
+    # Add cover to `e`, and remove `cover` from `rec`, because it's not
+    # a valid `Edition` field and would otherwise be copied in.
+
+    cover_url = rec.pop('cover', None)
+    if cover_url and not e.get_covers():
+        if cover_id := add_cover(cover_url, e.key, account_key=account_key):
+            e['covers'] = [cover_id]
+            need_edition_save = True
+
+    # Convert the record to an OpenLibrary Edition suitable for saving, so
+    # that values copied from the record into the Edition are of the proper
+    # format.
+    rec_edition = build_query(rec)
+
+    # Update `e` str and int value fields with `rec` values. Don't overwrite
```

Comment:
This comment says "`e` str and int value fields", but the code is checking `rec` values for str or int. 
Won't this match the author name string, which is a `str` on `rec`, but a `Thing` on `e`?

Reply:
I realized that comment was wrong. I used the wrong value name, and I need a better name for this, but I meant to write this is checking `rec_edition` for values, and here that comes from `build_query()`, so its `authors` value at this stage is something like `'authors': [<Author: '/authors/OL10A'>]`

In my testing this isn't copying the author into the edition.

I will update the comment.